sql/analyzer/resolve_column_defaults.go

```diff
@@ -208,7 +215,7 @@ func lookupColumnForTargetSchema(_ *sql.Context, node sql.SchemaTarget, colIndex
 		if colIndex < len(schema) {
 			return schema[colIndex], nil
 		} else {
-			return nil, sql.ErrColumnNotFound.New(colIndex)
+			return nil, expression.ErrIndexOutOfBounds.New(colIndex, len(schema))
```

Comment:
`sql.ErrColumnNotFound` might be more appropriate, we intentionally generate expression ids unrelated to schema ordering for uniqueness.